### ISSUE TITLE
Separate/Restructure the configuration files

### ISSUE THREAD

**Issue by h3xstream** [h3xstream/find-sec-bugs#77]
Currently, their are few configurations but one file is rapidly growing `methods-summaries.txt`.

I think it's a good moment to rethink how the API are stored logically[1] and the format of the file[2].
 - [1] List of files needed
 - [2] Maybe change the format to be more user friendly. (see reverse order of parameter and readability)

/cc @formanek, @naokikimura

**Comment by h3xstream** [comment]
# Configuration files distribution

findsecbugs/configurations/

 - injection-jdo.txt
 - injection-jpa.txt
 - injection-custom.txt
 - injection...
 - ta-safe-methods.txt
 - ta-tainted-methods.txt
 - ta-transfer-methods.txt
 - password-methods.txt

# Format

If we want to make it user-friendly, I think we should switch the parameters index reference to the "readable format" instead of the stack-based order. Most developer would say it is the reverse order.

Use pipe (`|`) as separator. I don't have strong opinion on the character to used. It is simply need to be visually different from the commonly used (`\`,`/`,`;`,`(` and `)`)

-----------------

That's it for me. Let me know what you think..

**Comment by formanek** [comment]
I was counting with splitting the file once it would be too large, so this can be done easily. I prefer splitting by API rather than by return taint state - it is easier to maintain and review. But if we do not modify Strings automatically, the size is still ok for me.

For parameters, you just need to know that they are counted from the right, from 0 and that long and double types are counted as two. We could reverse the order, but this would cause another confusion - are parameters counted from 0 or 1, what is the index of the object itself and are the parameters shifted for static methods? Change of separators is trivial.

To sum up, I think we could do something about this in the future, but now I have many improvements of a higher importance.

**Comment by h3xstream** [comment]
@formanek Ok I see your point for the both. I will bring little benifit.
I will close the ticket. You convince me.
